mqtt.cpp

```diff
@@ -71,24 +71,38 @@
 #endif
 
 extern OpenSprinkler os;
+extern char tmp_buffer[];
 
 #define MQTT_DEFAULT_PORT		1883	// Default port for MQTT. Can be overwritten through App config
 #define MQTT_MAX_HOST_LEN		50		// Note: App is set to max 50 chars for broker name
 #define MQTT_MAX_ID_LEN			16		// MQTT Client Id to uniquely reference this unit
 #define MQTT_RECONNECT_DELAY	120		// Minumum of 60 seconds between reconnect attempts
+#define MQTT_MAX_USERNAME_LEN 	50
```

Comment:
I think 20 should be sufficient for username and password

Reply:
Also worth preventing buffer overflow like I did here: https://github.com/OpenSprinkler/OpenSprinkler-Firmware/pull/133/files#diff-0f7914a36a22f82b449c1e46dcb60821R138